plutus-ledger-api/test/Spec/ScriptSize.hs

```diff
@@ -0,0 +1,203 @@
+    validatorV2 datumBuiltinData redeemerBuiltinData _scriptContext =
+      if expected == redeemer && expected == datum
+        then ()
+        else error ()
+      where
+        redeemer :: Integer
+        redeemer = V2.unsafeFromBuiltinData redeemerBuiltinData
+
+        datum :: Integer
+        datum = V2.unsafeFromBuiltinData datumBuiltinData
+
+codeV3 :: CompiledCode (BuiltinData -> BuiltinUnit)
+codeV3 = $$(compile [||validatorV3||])
+  where
+    validatorV3 :: BuiltinData -> BuiltinUnit
+    validatorV3 scriptContext =
+      if expected == redeemer && Haskell.Just expected == datum
+        then BI.unitval
+        else error ()
+      where
+        redeemer :: Integer
+        redeemer = V3.unsafeFromBuiltinData redeemerBuiltinData
+
+        datum :: Haskell.Maybe Integer
+        datum = V3.unsafeFromBuiltinData . V3.getDatum <$> optionalDatum
+
+        (redeemerBuiltinData, optionalDatum) =
+          case V3.unsafeFromBuiltinData scriptContext of
+            V3.ScriptContext
+              _txInfo
+              (V3.Redeemer redeemerBuiltinData')
+              (V3.SpendingScript _txOutRef optionalDatum') ->
+                (redeemerBuiltinData', optionalDatum')
+            _ -> error ()
+
+codeV3lazy :: CompiledCode (BuiltinData -> BuiltinUnit)
+codeV3lazy = $$(compile [||validatorV3smart||])
+  where
+    validatorV3smart :: BuiltinData -> BuiltinUnit
+    validatorV3smart scriptContext =
+      if expected == redeemer && expected == datum
+        then BI.unitval
+        else error ()
+      where
+        redeemerFollowedByScriptInfo :: BI.BuiltinList BuiltinData
+        redeemerFollowedByScriptInfo = BI.tail (constrArgs scriptContext)
+
+        redeemerBuiltinData :: BuiltinData
+        redeemerBuiltinData = BI.head redeemerFollowedByScriptInfo
+
+        scriptInfoData :: BuiltinData
+        scriptInfoData = BI.head (BI.tail redeemerFollowedByScriptInfo)
+
+        datumData :: BuiltinData
+        datumData = BI.head (constrArgs (BI.head (BI.tail (constrArgs scriptInfoData))))
+
+        redeemer :: Integer
+        redeemer = V3.unsafeFromBuiltinData redeemerBuiltinData
+
+        datum :: Integer
+        datum = V3.unsafeFromBuiltinData (V3.getDatum (V3.unsafeFromBuiltinData datumData))
+
+constrArgs :: BuiltinData -> BI.BuiltinList BuiltinData
+constrArgs = BI.snd . BI.unsafeDataAsConstr
+
+expected :: Integer
+expected = 42
+
+{-
+  Constr
+    0
+    [ Constr
+      0
+      [ List []
+      , List []
+      , List []
+      , I 1000000
+      , Map []
+      , List []
+      , Map []
+      , Constr 0
+        [ Constr 0 [Constr 0 []
+        , Constr 1 []]
+        , Constr 0 [Constr 2 []
+        , Constr 1 []]
+        ]
+      , List []
+      , Map []
+      , Map []
+      , B ""
+      , Map []
+      , List []
+      , Constr 1 []
+      , Constr 1 []
+      ]
+    , I 42
+    , Constr
+        1
+        [ Constr 0 [B "", I 100]
+        , Constr 0 [I 42]
+        ]
+    ]
+-}
```

Comment:
Did you get it via a GHCi session? In that case I'd include it to make it easy for the person updating this test to regenerate this output. Otherwise I'd probably just remove it, it's bound to get out of sync at some point.